main.go

```diff
@@ -89,9 +91,22 @@ func run(l *slog.Logger) error {
 	// create the namespace lister
 	nsl := NewNamespaceListerForSubject(accessCache)
 
+	// create metrics tracker
+	reg := NewDefaultRegistry()
+	metrics := NewMetrics(reg)
+
+	// serve metrics
+	metricsServer := NewMetricsServerBuilder().
+		WithAddress(metricsListenAddress).
+		WithTLS(enableTLS).
+		WithTLSConfig(loadTLSCert(l, tlsCertificatePath, tlsCertificateKeyPath)).
+		WithRegistry(reg).
+		Build()
+	go metricsServer.Serve(ctx)
```

Comment:
if the metrics server terminates, we may still want to call `cancel()` to be sure sure `ctx` is invalidated